beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/aws/S3Client.java

```diff
@@ -73,10 +74,21 @@ List<String> deleteObjects(String bucket, List<String> keys) {
       return Collections.emptyList();
     }
     if (!dryRunEnabled) {
-      keys.forEach(key -> log.info("Deleting: \"{}/{}\"", bucket, key));
-      DeleteObjectsRequest deleteObjectsRequest = new DeleteObjectsRequest(bucket)
-          .withKeys(keys.toArray(new String[] {}));
-      DeleteObjectsResult deleteObjectsResult = amazonS3.deleteObjects(deleteObjectsRequest);
+      log.info("Attempting to delete a total of {} objects, from [{}] to [{}]", keys.size(), keys.get(0), keys.get(keys.size()-1));
+      DeleteObjectsResult deleteObjectsResult = new DeleteObjectsResult(new ArrayList<>());
+      DeleteObjectsRequest deleteObjectsRequest = new DeleteObjectsRequest(bucket);
+      int totalKeys = keys.size();
+      int chunkSize = 1000;
+      int indexStart;
+      int indexEnd = 0;
+      while(indexEnd < totalKeys) {
+        indexStart = indexEnd;
+        indexEnd = indexStart + chunkSize < totalKeys ? indexStart + chunkSize : totalKeys;
+        deleteObjectsRequest .withKeys(keys.subList(indexStart, indexEnd).toArray(new String[] {}));
```

Comment:
What happens if one of these requests fails? Is it retried?

Reply:
At the moment there is no retry logic for this part of beekeeper, I see on some other part of the project we are using spring retry library, that could be an option. Alternatively i think i can add some specific retry logic by further breaking the request on smaller parts trying to delete as many files as possible and maybe find out if a specific file is failing to be deleted.

Reply:
Thanks, @RickArt. I think we should raise another issue to look at retry logic. As you said, there might be other parts that already have some retry mechanism in place.

Reply:
created a new ticket to track the feature

Reply:
Thanks. You could link it here for future reference.

Reply:
The created ticket is [EGDL-3574](https://jira.expedia.biz/browse/EGDL-3574)

Reply:
@RickArt – I deleted your comment. You can't link our tickets here because this is open source. You need to create an issue here in GitHub to track it and get it implemented.